tardis/plasma/properties/nlte.py

```diff
@@ -103,3 +105,103 @@ def filter_with_helium_index(atomic_number, ion_number, partition_function,
             columns=['ionization_energy'])
         return partition_function_index, ionization_data_index,\
                partition_function, ionization_data
+
+class HeliumNumericalNLTE(ProcessingPlasmaProperty):
+    outputs = ('helium_population',)
+    '''
+    IMPORTANT: This particular property requires a specific numerical NLTE
+    solver and a specific atomic dataset (neither of which are distributed
+    with Tardis) to work.
+    '''
+    def calculate(self, ion_number_density, electron_densities, t_electrons, w,
+        lines, j_blues, levels, level_boltzmann_factor, t_rad,
+        zeta_data, g_electron, delta, partition_function, ionization_data,
+        beta_rad, g):
+        print 'Performing numerical NLTE calculations.'
+        if len(j_blues)==0:
+            return None
+        heating_rate_data = np.loadtxt(
+            self.plasma_parent.heating_rate_data_file, unpack=True)
+        #Outputting data required by SH module
+        for zone in range(0, len(electron_densities)):
+            output_file = open('He_NLTE_Files/shellconditions_%d.txt' %zone,
+                'w')
+            print>>output_file, ion_number_density.ix[2].sum()[zone]
+            print>>output_file, electron_densities[zone]
+            print>>output_file, t_electrons[zone]
+            print>>output_file, heating_rate_data[zone]
+            print>>output_file, w[zone]
+            print>>output_file, self.plasma_parent.time_explosion
+            print>>output_file, t_rad[zone]
+            print>>output_file, self.plasma_parent.v_inner[zone]
+            print>>output_file, self.plasma_parent.v_outer[zone]
+            output_file.close()
+
+            output_file = open('He_NLTE_Files/abundances_%d.txt' %zone, 'w')
+            for element in range(1,31):
+                try:
+                    number_density = ion_number_density[zone].ix[element].sum()
+                except:
+                    number_density = 0.0
+                print>>output_file, number_density
+            output_file.close()
+
+            helium_lines = lines[lines['atomic_number']==2]
+            helium_lines = helium_lines[helium_lines['ion_number']==0]
+            output_file = open('He_NLTE_Files/discradfield_%d.txt' %zone, 'w')
+            j_blues = pd.DataFrame(j_blues, index=lines.index)
+            helium_j_blues = j_blues[zone].ix[helium_lines.index]
+            for value in helium_lines.index:
+                if (helium_lines.level_number_lower.ix[value]<35):
+                    print>>output_file, \
+                        int(helium_lines.level_number_lower.ix[value]+1), \
+                        int(helium_lines.level_number_upper.ix[value]+1), \
+                        j_blues[zone].ix[value]
+            output_file.close()
+        #Running numerical simulations
+        for zone in range(0, len(electron_densities)):
```

Comment:
in cases like these it is I think better to use

```Python
for zone, _ in enumerate(electron_densities):
    do stuff with zone
```
Replace underscore with a variable name if you need to use the value as well